### DISCUSSION TITLE
Can't understand how to read file

### DISCUSSION THREAD

**Discussion by cueckoo** [cue-lang/cue#886] [Q&A] [answered]
_Originally opened by @selfuryon in https://github.com/cuelang/cue/discussions/886_

Hello!
I found struct `Read` in `file` package, but I can't understand how to use it

**test.txt**
```
File content
```

**main.cue**
```
package main

import (
    "tool/file"
)

let ca = file.Read & {
    filename: "test.txt"
    contents: string
}

test: {
    contents: ca.contents
}
```

**attempt:**
```bash
λ cue eval
test: {
    contents: string
}
```

**Comment by cueckoo** [comment, marked as answer]
_Original reply by @mpvl in https://github.com/cuelang/cue/discussions/886#discussioncomment-581142_

Basically, anything in the `tool` directory is intended to be interpreted by `cue cmd` (see `cue help cmd`). To normal CUE this is just data. These methods are designed to be "dynamic" and thus are not hermetic and not suitable for normal CUE evaluation.

Note, though, that we are considering allowing statically importing non-CUE files, either through the import mechanism or through a special embed form.

**Reply by cueckoo** [reply]
_Original reply by @selfuryon in https://github.com/cuelang/cue/discussions/886#discussioncomment-581142_

Thanks!

**Reply by jzelinskie** [reply]
>Note, though, that we are considering allowing statically importing non-CUE files, either through the import mechanism or through a special embed form.

Is there an issue I can subscribe to for this?

**Reply by myitcv** [reply]
It's on my list to write up the embed proposal, @jzelinskie. Will try to remember to CC you that.
